beater/agent_config_handler.go

```diff
@@ -0,0 +1,106 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beater
+
+import (
+	"net/http"
+
+	"github.com/elastic/apm-server/convert"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/kibana"
+
+	"github.com/elastic/apm-server/agentcfg"
+)
+
+func agentConfigHandler(client *kibana.Client, secretToken string) http.Handler {
```

Comment:
nit: in some places you use `kb *kibana.Client` and here `client *kibana.Client`. It'd make it easier to grasp if we used the same variable names all along (I know we haven't focused on this so far, but newly introduced logic is always a good place to start improving).

Reply:
i could argue that context is different, too. a critical function in a big package can benefit from more explicit naming, like
`beater.newMuxer(beaterConfig *Config, client *kibana.Client, report publish.Reporter`

whether a small function in a narrow self-describing package can benefit more from concise naming, like `agentcfg.Fetch(kb *kibana.Client, q Query, err error)`

but I get your point

Reply:
Up to you if you want to change it or not. I see this really as nitpicking. 

I didn't intend to start a long discussion around this, but following up on your explanation:
I don't see how `kb` is _concise_, it is an abbreviation for the package `kibana` and doesn't address that it actually holds the `client` from this package. Following your argument about being explicit and concise I would head into the direction of:
`beater.newMuxer(beaterConfig *Config, kbClient *kibana.Client, report publish.Reporter)`: in the beater package you might have multiple different clients, so adding the package abbrev might be helpful
`agentcfg.Fetch(client *kibana.Client, q Query, err error)` not necessary to add the package info in the name, however it is still a `client` entity.